File: Python/Employee_and_Payroll_Manager/services/payroll_loader.py
import os
from models.employee import Employee

def load_employees(filepath, error_log_path):
    employees = []

    if not os.path.exists(filepath):
        print(f"Employee file not found: {filepath}")
        return employees

    os.makedirs(os.path.dirname(error_log_path), exist_ok=True)

    with open(filepath, "r") as f, open(error_log_path, "w") as error_log:
        for raw_line in f:
            line = raw_line.strip()

            if not line:
                continue 

            parts = line.split(",")

            if len(parts) != 4:
                error_log.write(f"Skipped (wrong number of fields): {line}\n")
                continue

            name, emp_id, salary_str, department = parts

            try:
                salary = float(salary_str)
            except ValueError:
                error_log.write(f"Skipped (invalid salary '{salary_str}'): {line}\n")
                continue

            employee = Employee(name, emp_id, salary, department)
            employees.append(employee)

    return employees

File: Employee_and_Payroll_Manager/Services/PayrollLoader.cs
using Models;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Services
{
    public static class PayrollLoader
    {
        public static List<Employee> LoadEmployees(string filepath, string errorLogPath)
        {
            List<Employee> employees = new List<Employee>();

            if (!File.Exists(filepath))
            {
                Console.WriteLine($"Employee file not found: {filepath}");
                return employees;
            }

            Directory.CreateDirectory(Path.GetDirectoryName(errorLogPath));

            using (StreamWriter errorLog = new StreamWriter(errorLogPath, false)) // false = overwrite
            {
                string[] lines = File.ReadAllLines(filepath);
                foreach(string rawLine in lines)
                {
                    string line = rawLine.Trim();

                    if (string.IsNullOrEmpty(line))
                    {
                        continue; // skip blank line silently
                    }

                    string[] parts = line.Split(',');

                    if(parts.Length != 4)
                    {
                        errorLog.WriteLine($"Skipped (wrong number of fields): {line}");
                        continue;
                    }

                    string name = parts[0];
                    string empId = parts[1];
                    string salaryStr = parts[2];
                    string department = parts[3];

                    if(!double.TryParse(salaryStr, out double salary))
                    {
                        errorLog.WriteLine($"Skipped (invalid salary '{salaryStr}'): {line}");
                        continue;
                    }

                    Employee employee = new Employee(name, empId, salary, department);
                    employees.Add(employee);
                }
            }
            return employees;
        }
    }
}
